src/lib.rs

```diff
@@ -64,14 +64,21 @@ fn get_metadata_raw(root: &State<String>, id: String) -> Result<String, OutpackE
 #[rocket::get("/file/<hash>")]
 pub async fn get_file(root: &State<String>, hash: String) -> Result<OutpackFile, OutpackError> {
     let path = store::file_path(&root, &hash);
-   OutpackFile::open(hash, path?).await
+    OutpackFile::open(hash, path?).await
         .map_err(|e| OutpackError::from(e))
 }
 
+#[rocket::get("/checksum")]
+pub async fn get_checksum(root: &State<String>) -> OutpackResult<String> {
+    metadata::get_ids_digest(&root)
```

Comment:
A couple of things here:

* I expect that we'll end up doing a generalised hash with this rust implementation later, so it might be worth taking a similar approach here to outpack itself to do the hash with a selectable algorithm and format the string as a helper function (e.g., https://github.com/mrc-ide/outpack/blob/main/R/hash.R#L14-L17)
* I agree that this does not need to be cryptographically secure but I think it should default to the hash in the config (which itself defaults to sha256 I think). It might also be worth allowing this algorithm to be overridden by a query parameter by the caller
* Can you document in the spec how the digest is computed as we discussed that this something that each outpack implementation might want to compute as it will be useful for cache invalidation everywhere